hedera-node/src/main/java/com/hedera/services/fees/calculation/consensus/txns/UpdateTopicResourceUsage.java

```diff
@@ -46,33 +46,36 @@ public boolean applicableTo(TransactionBody txn) {
-        try {
-            MerkleTopic merkleTopic = view.topics().get(MerkleEntityId.fromTopicId(txn.getConsensusUpdateTopic().getTopicID()));
-            long rbsIncrease = getUpdateTopicRbsIncrease(
-                    txn.getTransactionID().getTransactionValidStart(),
-                    JKey.mapJKey(merkleTopic.getAdminKey()),
-                    JKey.mapJKey(merkleTopic.getSubmitKey()),
-                    merkleTopic.getMemo(),
-                    merkleTopic.hasAutoRenewAccountId(),
-                    lookupExpiry(merkleTopic),
-                    txn.getConsensusUpdateTopic());
-            return getConsensusUpdateTopicFee(txn, rbsIncrease, sigUsage);
-        } catch (Exception illegal) {
-            log.warn("Usage estimation unexpectedly failed for {}!", txn, illegal);
-            throw new InvalidTxBodyException(illegal);
+    public FeeData usageGiven(final TransactionBody txnBody, final SigValueObj sigUsage, final StateView view) throws InvalidTxBodyException {
+        if (txnBody == null || !txnBody.hasConsensusUpdateTopic()) {
+            throw new InvalidTxBodyException("consensusUpdateTopic field not available for Fee Calculation");
         }
-    }
+        if (view == null) {
+            throw new IllegalStateException("No StateView present !!");
+        }
+
+        long rbsIncrease = 0;
+        MerkleTopic merkleTopic = view.topics().get(
+                MerkleEntityId.fromTopicId(txnBody.getConsensusUpdateTopic().getTopicID()));
 
-    private Timestamp lookupExpiry(MerkleTopic merkleTopic) {
-        try {
-            return Timestamp.newBuilder()
-                    .setSeconds(merkleTopic.getExpirationTimestamp().getSeconds())
-                    .build();
-        } catch (NullPointerException e) {
-			/* Effect is to charge nothing for RBH */
-            log.warn("Missing topic expiry data, ignoring expiration time in fee calculation!", e);
-            return ZERO_EXPIRY;
+        if (merkleTopic != null && merkleTopic.hasAdminKey()) {
+            try{
+                var expiry = Timestamp.newBuilder()
+                        .setSeconds(merkleTopic.getExpirationTimestamp().getSeconds())
+                        .build();
+                rbsIncrease = getUpdateTopicRbsIncrease(
+                        txnBody.getTransactionID().getTransactionValidStart(),
+                        JKey.mapJKey(merkleTopic.getAdminKey()),
+                        JKey.mapJKey(merkleTopic.getSubmitKey()),
+                        merkleTopic.getMemo(),
+                        merkleTopic.hasAutoRenewAccountId(),
+                        expiry,
+                        txnBody.getConsensusUpdateTopic());
+            } catch (DecoderException illegal) {
+                log.warn("Usage estimation unexpectedly failed for {}!", txnBody, illegal);
+                throw new InvalidTxBodyException(illegal);
+            }
         }
+        return getConsensusUpdateTopicFee(txnBody, rbsIncrease, sigUsage);
```

Comment:
Seems to me the `return` should only happen when the topic is found and its adminKey exists. Otherwise, it's better to throw an `IllegalStateException`.

Reply:
When an update Transaction is submitted with a missing topic or an immutable topic, the fee should be calculated as if the rbsIncrease is 0 as we are not changing anything.

Reply:
I see.